src/core/tools/base/types.ts

```diff
@@ -0,0 +1,7 @@
+export type ToolInput<T> = T
```

Comment:
I think we should put all backend types somewhere, right now they are a bit scattered.